neo/Prompt/Commands/Show.py

```diff
@@ -0,0 +1,159 @@
+from neocore.UInt256 import UInt256
+from neo.IO.MemoryStream import StreamManager
+from neo.Network.NodeLeader import NodeLeader
+from neo.Implementations.Notifications.LevelDB.NotificationDB import NotificationDB
+from neo.logging import log_manager
+import json
+
+
+logger = log_manager.getLogger()
+
+
+class CommandShow(CommandBase):
+    def __init__(self):
+        super().__init__()
+
+        self.register_sub_command(CommandShowBlock())
+        self.register_sub_command(CommandShowHeader())
+        self.register_sub_command(CommandShowTx())
+        self.register_sub_command(CommandShowMem())
+        self.register_sub_command(CommandShowNodes(), ['node'])
+
+    def command_desc(self):
+        return CommandDesc('show', 'show data from the blockchain')
+
+    def execute(self, arguments):
+        item = get_arg(arguments)
+
+        if not item:
+            print("run `%s help` to see supported queries" % CommandShow().command_desc().command)
+            return
+
+        try:
+            return self.execute_sub_command(item, arguments[1:])
+        except KeyError:
+            print(f"{item} is an invalid parameter")
+            return
+
+
+class CommandShowBlock(CommandBase):
+    def __init__(self):
+        super().__init__()
+
+    def execute(self, arguments):
+        item = get_arg(arguments)
+        txarg = get_arg(arguments, 1)
+
+        block = Blockchain.Default().GetBlock(item)
+
+        if block is not None:
+            block.LoadTransactions()
+
+            if txarg and 'tx' in txarg:
+                txs = []
+                for tx in block.FullTransactions:
+                    print(json.dumps(tx.ToJson(), indent=4))
+                    txs.append(tx.ToJson())
+                return txs
+
+            print(json.dumps(block.ToJson(), indent=4))
+            return block.ToJson()
+
+        else:
+            print("Could not locate block %s" % item)
+            return
+
+    def command_desc(self):
+        p1 = ParameterDesc('index/hash', 'the index or scripthash of the block')
+        p2 = ParameterDesc('tx', 'arg to only show block transactions', optional=True)
+        return CommandDesc('block', 'show a specified block', [p1, p2])
+
+
+class CommandShowHeader(CommandBase):
+    def __init__(self):
+        super().__init__()
+
+    def execute(self, arguments):
+        item = get_arg(arguments)
+
+        header = Blockchain.Default().GetHeaderBy(item)
+        if header is not None:
+            print(json.dumps(header.ToJson(), indent=4))
+            return header.ToJson()
+        else:
+            print("Could not locate header %s\n" % item)
+            return
+
+    def command_desc(self):
+        p1 = ParameterDesc('index/hash', 'the index or scripthash of the block header')
+        return CommandDesc('header', 'show the header of a specified block', [p1])
+
+
+class CommandShowTx(CommandBase):
+    def __init__(self):
+        super().__init__()
+
+    def execute(self, arguments):
```

Comment:
show tx with an invalid argument gives a not user friendly error
```
neo> show tx 1
Could not find transaction from args: Invalid UInt256 Format: 1 chars != 64 chars (['1'])
```

Reply:
How about now? (I just removed the exception)